maas/resource_maas_instance_test.go

```
@@ -0,0 +1,211 @@
+			},
+			// Test update
+			{
+				Config: testAccMAASInstanceConfigSetup(vmHost, hostname) + testAccMAASInstanceConfigReleaseParams(comment, erase, force, quickErase, secureErase),
+				Check: resource.ComposeTestCheckFunc(append(
+					baseChecks,
+					resource.TestCheckResourceAttr("maas_instance.test", "release_params.#", "1"),
+					resource.TestCheckResourceAttr("maas_instance.test", "release_params.0.comment", comment),
+					resource.TestCheckResourceAttr("maas_instance.test", "release_params.0.erase", erase),
+					resource.TestCheckResourceAttr("maas_instance.test", "release_params.0.force", force),
+					resource.TestCheckResourceAttr("maas_instance.test", "release_params.0.quick_erase", quickErase),
+					resource.TestCheckResourceAttr("maas_instance.test", "release_params.0.secure_erase", secureErase),
+				)...,
+				),
+			},
+			// Test destroy leaves the machine in a ready state
+			{
+				Config: testAccMAASInstanceConfigSetup(vmHost, hostname),
+				Check:  testAccMAASInstanceCheckMachineLogsForDestroy(hostname, erase == "true"),
+			},
+		},
+	})
+}
+
+// Check logs for relevant events to determine if the machine was released as expected during destroy
+func testAccMAASInstanceCheckMachineLogsForDestroy(hostname string, erase bool) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		conn := testutils.TestAccProvider.Meta().(*maas.ClientConfig).Client
+
+		params := entity.EventParams{
+			Hostname: hostname,
+		}
+
+		events, err := conn.Events.Get(&params)
+		if err != nil {
+			return err
+		}
+
+		if len(events.Events) == 0 {
+			return fmt.Errorf("no events found for hostname %s", hostname)
+		}
+
+		// Check through all events to see if the machine was released as expected
+		wasErased := false
+		wasReleased := false
+
+		for _, event := range events.Events {
+			if event.Type == "Disks erased" {
+				wasErased = true
+			}
+
+			if event.Type == "Released" {
+				wasReleased = true
+			}
```

you could probably also exit early here with something like `if (wasReleased && wasErased) { break }`
Not that there will be a huge performance hit probably if you do look at the rest of the event log

---

I initially agreed but actually realised I have to check the entire log, because if the machine has been erased when it shouldn't be expected, then this should throw an error. 

Thinking about this more thoroughly did catch an additional error in this test which I have just fixed